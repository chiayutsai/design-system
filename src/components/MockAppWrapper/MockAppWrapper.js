import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
