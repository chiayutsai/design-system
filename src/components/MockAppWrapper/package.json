{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
