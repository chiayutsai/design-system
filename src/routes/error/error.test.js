import renderer from 'react-test-renderer'
import MockAppWrapper, { store } from 'components/MockAppWrapper'
import setupLocale from 'test/utils/setupLocale'
import action from './index'

describe('error', () => {
  // setup i18next if component needed
  setupLocale(store)

  test('render correctly', async () => {
    const query = {}

    const { component } = await action({ query })

    const wrapper = renderer
      .create(<MockAppWrapper>{component}</MockAppWrapper>)
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
