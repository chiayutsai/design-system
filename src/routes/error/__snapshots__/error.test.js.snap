// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error render correctly 1`] = `
Array [
  <h1>
    Error 错误
  </h1>,
  <p>
    系统错误，请稍后再试。[
    99999
    ]
  </p>,
  <p>
    System error, please try again later.[
    99999
    ]
  </p>,
]
`;
